core/launcher/src/manager/app_launcher.rs

```diff
@@ -513,8 +515,43 @@ impl AppLauncher {
         }
     }
 
-    fn get_transport(url: String) -> AppRuntimeTransport {
-        if !url.as_str().contains("__firebolt_endpoint") {
+    fn get_modified_url(manifest: AppManifest, sessionid: &str) -> String {
```

Comment:
I don't think we need an owned manifest. Cloning is a costly operation. We have to avoid it as much as possible. Here in this function, we are only reading the values from manifest. So I think we can take that as a reference.

Reply:
done